src/ai/backend/manager/api/session.py

```diff
@@ -2179,6 +2181,92 @@ async def shutdown_service(request: web.Request, params: Any) -> web.Response:
     return web.Response(status=204)
 
 
+async def find_dependency_sessions(
+    root_session_name_or_id: Union[uuid.UUID, str],
+    db_connection: SAConnection,
+    access_key: AccessKey,
+):
+    @aiotools.lru_cache(maxsize=100)
+    async def _find_dependency_sessions(session_name_or_id: Union[uuid.UUID, str]):
```

Comment:
Adding `lru_cache()` decorator to inner function will be meaningless since this function will be redeclared every time outer function is called. Please declare it as a separate outer function.